ypkg2/main.py

```diff
@@ -190,6 +191,41 @@ def execute_step(context, step, step_n, work_dir):
             return False
     return True
 
+was_avx2_context = False
+
+def post_execute_step(context, step, step_n, work_dir):
+
+    hwcaps_v3_bin_dir = os.path.join(context.get_install_dir(), "usr", "hwcaps", "x86-64-v3", "bin")
+    hwcaps_v1_bin_dir = os.path.join(context.get_install_dir(), "usr", "hwcaps", "x86-64-v1", "bin")
+    bin_dir = os.path.join(context.get_install_dir(), "usr", "bin")
+
+    if context.avx2 and step_n == "install":
+        global was_avx2_context
+        was_avx2_context = True
+
+        os.makedirs(hwcaps_v3_bin_dir, exist_ok=True)
+        for dirpath, dirnames, filenames in os.walk(bin_dir):
+            for file in filenames:
+                file_path = os.path.join(dirpath, file)
+                if is_elf_file(file_path):
+                    shutil.move(file_path, hwcaps_v3_bin_dir)
+
+    if context.emul32 is False and context.avx2 is False and step_n == "install" and was_avx2_context is True:
```

Comment:
It would be great if we could also inverse all of this for an early return. There are a lot of elements in this check as well that make it a bit hard to read.